### PR TITLE
refactor(ui): remove TSFixMes and replace with correct types

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- remove TSFixMes and replace with correct types.

## QA

N/A

### CODE REVIEW

File: ui/src/app/machines/views/MachineDetails/MachineTests/MachineTestsTable/TestActions/TestActions.tsx [hunk 1]
```diff
@@ -19,8 +21,7 @@ const TestActions = ({ scriptResult, setExpanded }: Props): JSX.Element => {
   const location = useLocation();
   const canViewDetails = !scriptResultInProgress(scriptResult.status);
   const hasMetrics = scriptResult.results.length > 0;
-  // TODO - Update ContextualMenu props to be able to accept non Button props
-  const links: TSFixMe = [];
+  const links: ContextualMenuProps<DataTestElement<LinkProps>>["links"] = [];
```
Comment (Caleb-Ellis) [inline ui/src/app/machines/views/MachineDetails/MachineTests/MachineTestsTable/TestActions/TestActions.tsx:24]: Getting a bit nested here 😬

File: ui/src/app/store/script/utils.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import type { AnyObject } from "app/base/types";
+
+// The script parameters are parsed from a JSON object with any shape so we have
+// to do a bit more work to be sure we're retrieving valid values.
+export const getParameterObject = (
```
Comment (Caleb-Ellis) [inline ui/src/app/store/script/utils.ts:5]: I feel like `getObjectValue` and `getObjectString` might be more descriptive function names. I'm never really sure what a "parameter" is?

File: ui/src/app/store/script/utils.test.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    });
+
+    it("can handle parameters of the wrong type", () => {
+      const obj = { unknown: true };
+      expect(getParameterObject(obj.unknown, "object")).toBeNull();
+    });
+  });
+
+  describe("getStringParameter", () => {
+    it("can get a parameter object", () => {
```
Comment (Caleb-Ellis) [inline ui/src/app/store/script/utils.test.ts:24]: should be "string"

**Review by Caleb-Ellis — APPROVED** [review]
LGTM! Just a comment on function names

File: ui/src/app/machines/views/MachineDetails/MachineTests/MachineTestsTable/TestActions/TestActions.tsx (hunk 1 above)
Comment (huwshimi) [reply ui/src/app/machines/views/MachineDetails/MachineTests/MachineTestsTable/TestActions/TestActions.tsx:24]: Thanks, I've moved the `DataTestElement<LinkProps>` to a type so it reads a bit better.

File: ui/src/app/store/script/utils.ts (hunk 2 above)
Comment (huwshimi) [reply ui/src/app/store/script/utils.ts]: Done, I was trying to capture that these were for use with the `ScriptsParameters` objects, but the generic name is fine.

File: ui/src/app/store/script/utils.test.ts (hunk 3 above)
Comment (huwshimi) [reply ui/src/app/store/script/utils.test.ts:24]: Thanks, done.

**Pull request merged**